pkg/kubectl/describe.go

```diff
@@ -1558,12 +1587,14 @@ func (d *NodeDescriber) Describe(namespace, name string) (string, error) {
 	}
 
 	var events *api.EventList
-	if ref, err := api.GetReference(node); err != nil {
-		glog.Errorf("Unable to construct reference to '%#v': %v", node, err)
-	} else {
-		// TODO: We haven't decided the namespace for Node object yet.
-		ref.UID = types.UID(ref.Name)
-		events, _ = d.Events("").Search(ref)
+	if describerSetting.ShowEvents {
+		if ref, err := api.GetReference(node); err != nil {
+			glog.Errorf("Unable to construct reference to '%#v': %v", node, err)
+		} else {
+			// TODO: We haven't decided the namespace for Node object yet.
```

Comment:
At start I was wondering if there are any plans for making nodes namescoped. Then I searched and found out https://github.com/kubernetes/kubernetes/issues/4879. Add it under this TODO.